src/API/Site/Controllers/MerchantCenter/AttributeMappingCacheController.php

```diff
@@ -0,0 +1,103 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseOptionsController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\TransientsInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use WP_REST_Request as Request;
+use WP_REST_Response as Response;
+use Exception;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class for handling API requests handling cache for Attribute Mapping
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter
+ */
+class AttributeMappingCacheController extends BaseOptionsController {
+
+	/**
+	 * @var TransientsInterface
+	 */
+	private TransientsInterface $transients;
+
+
+	/**
+	 * AttributeMappingCacheController constructor.
+	 *
+	 * @param RESTServer          $server
+	 * @param TransientsInterface $transients
+	 */
+	public function __construct( RESTServer $server, TransientsInterface $transients ) {
+		parent::__construct( $server );
+		$this->transients = $transients;
+	}
+
+	/**
+	 * Register rest routes with WordPress.
+	 */
+	public function register_routes(): void {
+		/**
+		 * DELETE Flush the cache
+		 */
+		$this->register_route(
+			'mc/mapping/cache',
```

Comment:
This is (as discussed) for us to clear the cache after entering into the page.  So I was thinking about calling this on the frontend.

If you know any way to hook when the page is visited in the backend could be even better

Reply:
From my understanding we discussed that the clear on page visit would only be needed if the cache expiry was long. But if we are sticking with a short expiry (1 hour) then we don't need to trigger it.
They would only end up with stale data if they open up the mapping rules, add new meta keys for their products and then move back to the mapping rules. Which doesn't sound like a common scenario, and if it would happen it would eventually correct itself or a tool to clear transients can be used.

Reply:
Ah ok, I didn't catch that one. However, now we deleted all the metas, so that endpoint is not needed anymore.